### PR TITLE
Fix iOS HLS issues (probably)

### PR DESCRIPTION
**Pull request by netpro2k**
Fix video aspect ratio being broken, fix video being flipped, and an attempted fix for detecting live/VOD state (though I was not able to test this locally).

### CODE REVIEW

File: src/components/media-views.js [hunk 1]
```diff
@@ -174,7 +163,7 @@ function createVideoTexture(url, contentType) {
       videoEl.onerror = reject;
     }
 
-    videoEl.addEventListener("loadedmetadata", () => resolve(texture), { once: true });
+    videoEl.addEventListener("canplay", () => resolve(texture), { once: true });
```
Comment (netpro2k) [inline src/components/media-views.js]: Video size was not set yet when the `loadedmetadata` event fired for HLS videos. This is probably a more "correct" event for us to wait on for everything anyway

File: src/components/media-views.js [hunk 2]
```diff
@@ -156,16 +155,6 @@ function createVideoTexture(url, contentType) {
       } else if (videoEl.canPlayType(contentType)) {
         videoEl.src = url;
         videoEl.onerror = reject;
-
-        // HACK aframe iOS HLS video hacks
-        if (isIOS) {
```
Comment (netpro2k) [inline src/components/media-views.js]: Looks like these hacks only break things, and things seem fine without them or any sort of special shader. Chalk it up to iOS updates?

File: src/components/media-views.js [hunk 3]
```diff
@@ -499,13 +493,12 @@ AFRAME.registerComponent("media-video", {
         if (texture.hls.currentLevel >= 0) {
           updateLiveState();
         }
+      } else {
+        this.videoIsLive = this.video.duration === Infinity;
```
Comment (netpro2k) [inline src/components/media-views.js]: No clue if this is actually going to work, but I can't get a streaming source to work locally to try it out.

File: src/components/media-views.js (hunk 2 above)
Comment (johnshaughnessy) [reply src/components/media-views.js]: down with the HACK

**Pull request merged**